SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -135,5 +157,49 @@ public override bool InstanceEnabled
                 }
             }
         }
+
+        private void ProcessPendingErrors()
+        {
+            foreach (var file in ErrorLogHelper.GetErrorLogFiles())
+            {
+                AppCenterLog.Debug(LogTag, $"Process pending error file {file.Name}");
+                var log = ErrorLogHelper.ReadErrorLogFile(file);
+                if (log == null)
+                {
+                    AppCenterLog.Error(LogTag, $"Error parsing error log. Deleting invalid file: {file.Name}");
+                    try
+                    {
+                        file.Delete();
+                    }
+                    catch (System.Exception ex)
+                    {
+                        AppCenterLog.Warn(LogTag, $"Failed to delete error log file {file.Name}.", ex);
+                    }
+                }
+                else
+                {
+                    _unprocessedManagedErrorLogs.Add(log.Id, log);
+                }
+            }
+            SendCrashReportsOrAwaitUserConfirmation();
+        }
+
+        private void SendCrashReportsOrAwaitUserConfirmation()
+        {
+            HandleUserConfirmation();
+        }
+
+        private void HandleUserConfirmation()
+        {
+
+            /* Send every pending log. */
+            var keys = _unprocessedManagedErrorLogs.Keys.ToList();
+            foreach (var key in keys)
+            {
+                Channel.EnqueueAsync(_unprocessedManagedErrorLogs[key]);
+                _unprocessedManagedErrorLogs.Remove(key);
```

Comment:
We are accessing this and modifying it without locking or any thread safety, I think.

Reply:
This will eventually require locking to be thread-safe. At this time for the scope of the implementation, there is no threading issues.